### ISSUE TITLE
on('notification') not firing on iOS

### ISSUE THREAD

**Issue by bromeostasis** [phonegap/phonegap-plugin-push#1959]
### Expected Behaviour

After I tap a push notification, I should be able to handle the event and do things in my javascript code.

### Actual Behaviour

The app opens, but no events are fired.

### Reproduce Scenario (including but not limited to)

I have found this on iOS 10, iOS 11, plugin v1.9.0, and v1.10.5. I believe this started happening after I upgraded from cordova 5.3.3 to cordova 7.0.0.

#### Steps to Reproduce 

1. Send a push notification on iOS
2. Tap it
3. Look in a debugger for any type of logs/events being fired

Just to clarify, I've been using this plugin with great success for at least a year up to this point. I've had everything working and functional. I changed nothing in our js code, only plugin versions and cordova versions and then this problem arose.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

- iOS 11 beta
- iOS 10

#### Cordova CLI version and cordova platform version

cordova version 7.0.0
ios platform version 4.3.0

#### Plugin version

Reproducable for us on 1.9.0 and 1.10.5. I can try other versions if necessary.

#### Sample Push Data Payload

At first, I thought it was because I'd set content-enabled:true, but I've now tested this with even the simplest suggested payload. Even this did not work.

I also tried pushes from both sandbox and production.

```
{
  "from":"node-apn",
  "source":"web",
  "aps":{
    "alert":"Hello, world!"
  }
}
```

#### Sample Code that illustrates the problem

```
            push = PushNotification.init({
                "ios": {
                    "alert": "true",
                    "badge": "true",
                    "sound": "true"
                },
                "android": {
                    "icon": "",
                    "senderID": "",
                    "clearNotifications": false
                }
            });

            push.on('registration', function(data) {

                serverapi.registerPushToken(data.registrationId, device.platform).done(function(data) {
                    console.log("Registered push token!")
                }).fail(function() {
                    console.log("Failed to register push token")
                })

            });

            push.on('notification', function(data) {
                notiMessage = data.message;
                console.log(notiMessage) //
            });

            push.on('error', function(e) {
                console.log(e.message);
            });
```

#### Logs taken while reproducing problem

None really available. I did try to set up another push.on('notification') in the safari console and received an invalid JSON error, but it was not anything useful. If there are any logs you think I could find, they'd be super useful, but XCode device debugger is pretty much crap 😆 .

**Comment by agil-NUBBA** [comment]
Same, the function is not being triggered but the console prints:

```
didReceiveNotification with fetchCompletionHandler
app active
Notification received
```

**Comment by bromeostasis** [comment]
@agil-NUBBA are those your custom `js` logs? Just looking for any information I can get on this 😄

**Comment by bromeostasis** [comment]
I'm now able to reproduce this on `2.0.0` as well. Installed the latest version of the plugin and there is *no trace* of the event being fired. I'm happy to try **anything** that might help get some information.

@macdonst I'd argue that this is not a "question" as you've tagged, but a serious bug! I'm on the latest cordova, iOS, and XCode setup and I can't get the event to fire. I really need to get this working as a lot of our clients rely on this ability to redirect on tapping a push notification.

Thanks!!

**Comment by bromeostasis** [comment]
Trying out v 1.8.4, as that was working before the cordova 7 upgrade. Still no success, but got a few more lines of logging:

```
2017-09-15 16:36:23.445653-0400 CommCASH[1351:293155] [error] Activity feed result is not NSDictionary!
2017-09-15 16:36:23.470896-0400 CommCASH[1351:292252] [error] error:  (NSFetchedResultsController): couldn't read cache file to update store info timestamps
CoreData: error:  (NSFetchedResultsController): couldn't read cache file to update store info timestamps
2017-09-15 16:36:23.657396-0400 CommCASH[1351:292252] active
2017-09-15 16:36:23.657536-0400 CommCASH[1351:292252] PushPlugin skip clear badge
```

**Comment by bromeostasis** [comment]
@agil-NUBBA @macdonst finally got it figured out. Turns out there was an incompatibility with this plugin and the specific version of [the local notification plugin](https://github.com/katzer/cordova-plugin-local-notifications) that was preventing me from handling this event correctly.

Specifically, I was using `cordova plugin add https://github.com/katzer/cordova-plugin-local-notifications.git#0.8.5 --nofetch`. This version showed local notifications in the foreground, which I'm assuming was the cause of the issues, since it also caused the push plugin to show notifications in the foreground.

I switched to `https://github.com/Telerik-Verified-Plugins/LocalNotification.git` and the problem was solved.

Let me know if you'd like any more details or if you want this documented somewhere else! 😅
